testsuite/tests/ui/mail_and_messages/test_automatic_mails.py

```diff
@@ -44,3 +67,48 @@ def test_invitation_mail(account, navigator, mailhog_client, provider_account, r
     custom_devel_login(name=username, password=password)
 
     assert "SIGNED IN SUCCESSFULLY" in browser.element('//*[@id="flash-messages"]/div/div').text
+
+
+# pylint: disable=too-many-arguments
+@pytest.mark.usefixtures("login")
+def test_mail_application_plan_changed(extra_plan, navigator, service, custom_app, mailhog_client):
+    """
+    Test:
+        - Change application plan
+        - Assert that API consumer is notified about plan change
+        - Assert that API provider is notified about plan change
+    """
+    application = custom_app()
+    app = navigator.navigate(ApplicationDetailView, application=application, product=service)
+    app.change_plan(extra_plan["name"])
+
+    mailhog_client.assert_message_received(
+        subject=f"Application plan changed to '{extra_plan['name']}'", expected_count=1
+    )
+    mailhog_client.assert_message_received(
+        subject=f"Application {application.entity_name} has changed to plan {extra_plan['name']}", expected_count=1
+    )
+
+
+def test_mail_request_app_plan_changed(extra_plan, account, navigator, custom_devel_login, custom_app, mailhog_client):
```

Comment:
but `# pylint: disable=too-many-arguments` required here, not visible because of a "problem" with line 34